modules/pipeline/endpoints/queue_management.go

```diff
@@ -176,3 +176,38 @@ func (e *Endpoints) deletePipelineQueue(ctx context.Context, r *http.Request, va
 
 	return httpserver.OkResp(nil)
 }
+
+func (e *Endpoints) batchUpgradePipelinePriority(ctx context.Context, r *http.Request, vars map[string]string) (httpserver.Responser, error) {
+	var req apistructs.PipelinePriorityBatchUpgradeRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		return apierrors.ErrUpgradePipelinePriority.InvalidParameter(fmt.Errorf("failed to unmarshal request body, err: %v", err)).ToResp(), nil
+	}
+
+	identityInfo, err := user.GetIdentityInfo(r)
+	if err != nil {
+		return errorresp.ErrResp(err)
+	}
+	if !identityInfo.IsInternalClient() {
+		return apierrors.ErrUpgradePipelinePriority.AccessDenied().ToResp(), nil
+	}
+
+	if req.QueueID == 0 {
+		return nil, apierrors.ErrUpgradePipelinePriority.InvalidParameter("queueID")
+	}
+	if len(req.PipelineIDs) == 0 {
+		return nil, apierrors.ErrUpgradePipelinePriority.InvalidParameter("pipelineIDs")
+	}
+
+	queue, err := e.queueManage.GetPipelineQueue(req.QueueID)
+	if err != nil {
+		return errorresp.ErrResp(err)
+	}
+
+	queue.Usage = e.reconciler.QueueManager.QueryQueueUsage(queue)
+
+	if err = e.queueManage.BatchUpgradePipelinePriority(queue, req.PipelineIDs, e.reconciler); err != nil {
```

Comment:
put BatchUpgradePipelinePriority into reconciler.queueManager.

e.queueManage is a simple service wrapper for db.

Reply:
done